### PR TITLE
Add output_basename

### LINKED ISSUE #438: add file naming arguments to sample()
`save_output_files()` has several useful file naming arguments that are great for helping the user craft meaningful file names. However, only one of these arguments (`output_dir`) can be specified when saving sampling output directly from `sample()`. The remaining arguments use the function defaults.

I'd like to see arguments added to `sample()` that allowing the user to specify the `basename`, `timestamp`, and `random` arguments. I think this would make the output CSVs much more human-readable. To me, the current solution to use `output_dir` if you want the default naming, but needing a separate function call to change one argument seems inconsistent.

I think it would be most intuitive to pass `sample()` a list of arguments that are used to save the output CSVs. However, the current convention seems to avoid this type of grouping (compare with `rstan::sampling()`'s `control`). Directly adding the arguments may be the least intrusive solution.

Kudos for developing such a great package for interfacing Stan!

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Summary

Fixes #438 by adding an argument `output_basename`.

```
> fit <- mod$sample(data = data_file, chains = 4, output_basename = "custom")
> print(fit$output_files())
[1] "/tmp/RtmpHcV04N/custom-1.csv" "/tmp/RtmpHcV04N/custom-2.csv" "/tmp/RtmpHcV04N/custom-3.csv" "/tmp/RtmpHcV04N/custom-4.csv"
```
Combine that with output_dir to have completely predictable output CSV filepaths.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
Rok Češnovar

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
@wpetry @seth127 mind looking at this, if this works for you?

```
remotes::install_github("stan-dev/cmdstanr@output_basename"
```
to try it out.

File: man-roxygen/model-common-args.R [hunk 1]
```diff
@@ -48,15 +48,19 @@
...
+#'   methods of the fitted model object (e.g.,
+#'   [`$save_output_files()`][fit-method-save_output_files]). These temporary
+#'   files are removed when the fitted model object is
+#'   [garbage collected][base::gc] (manually or automatically).
+#'   * If a path, then the files are created in `output_dir` with names
+#'   corresponding to the defaults used by `$save_output_files()`.
 #'
+#' @param output_basename (string) A string to use as a prefix for the
+#'   names of the output CSV files of CmdStan.
+#'   * If `NULL` (the default), the
```
Comment (jgabry) [inline man-roxygen/model-common-args.R:62]: Did you mean to include `* If NULL (the default), the`? It just cuts off after that.

**Comment by jgabry** [comment]
This looks good, I just made one comment on the doc

File: man-roxygen/model-common-args.R (hunk 1 above)
Comment (rok-cesnovar) [reply man-roxygen/model-common-args.R:62]: Yeah, forgot to finish my sentance :)

Comment (rok-cesnovar) [reply man-roxygen/model-common-args.R]: Fixed.

**Comment by wpetry** [comment]
Looks good to me too. Thanks for implementing this feature!

**Pull request merged**